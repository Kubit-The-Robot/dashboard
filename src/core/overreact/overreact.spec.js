import OverReact from './index';

describe('createElement', () => {
  it('should return a DOM object', () => {
    const element = OverReact.createElement('div');

    expect(element).toEqual({
      type: 'div',
      props: { children: [] },
    });
  });

  it('should return a DOM object with an props and children', () => {
    const element = OverReact.createElement(
      'div',
      { id: 'foo' },
      OverReact.createElement('strong', null, 'bar'),
      OverReact.createElement('Oi, eu sou um texto')
    );

    expect(element).toEqual({
      type: 'div',
      props: {
        id: 'foo',
        children: [
          {
            type: 'strong',
            props: {
              children: [
                {
                  type: 'TEXT_ELEMENT',
                  props: { nodeValue: 'bar', children: [] },
                },
              ],
            },
          },
          { type: 'Oi, eu sou um texto', props: { children: [] } },
        ],
      },
    });
  });
});
