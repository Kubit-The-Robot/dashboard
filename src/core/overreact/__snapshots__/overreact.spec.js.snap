// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement should return a DOM object with an array of DOM objects 1`] = `
Object {
  "props": Object {
    "children": Array [
      Object {
        "props": Object {
          "children": Array [
            Object {
              "props": Object {
                "children": Array [],
                "nodeValue": "Texto em negrito",
              },
              "type": "TEXT_ELEMENT",
            },
          ],
        },
        "type": "strong",
      },
      Object {
        "props": Object {
          "children": Array [
            Object {
              "props": Object {
                "children": Array [],
                "nodeValue": "Text em it√°lico",
              },
              "type": "TEXT_ELEMENT",
            },
          ],
        },
        "type": "i",
      },
    ],
  },
  "type": "div",
}
`;

exports[`createElement should return a DOM object with an props and children 1`] = `
Object {
  "props": Object {
    "children": Array [
      Object {
        "props": Object {
          "children": Array [],
          "nodeValue": "Oi, eu sou um texto",
        },
        "type": "TEXT_ELEMENT",
      },
    ],
  },
  "type": "div",
}
`;

exports[`createElement should return an object of type div without child elements 1`] = `
Object {
  "props": Object {
    "children": Array [],
  },
  "type": "div",
}
`;
